v2/controllers/marketplace/meterbase_controller.go

```diff
@@ -1167,24 +1386,42 @@ func (r *MeterBaseReconciler) uninstallPrometheus(
 	}
 
 	return append(actions,
-		HandleResult(
-			GetAction(types.NamespacedName{Namespace: sm.Namespace, Name: sm.Name}, sm),
-			OnContinue(DeleteAction(sm))),
-		HandleResult(
-			GetAction(types.NamespacedName{Namespace: service2.Namespace, Name: service2.Name}, deployment),
-			OnContinue(DeleteAction(service2))),
 		HandleResult(
 			GetAction(types.NamespacedName{Namespace: service.Namespace, Name: service.Name}, service),
 			OnContinue(DeleteAction(service))),
-		HandleResult(
-			GetAction(types.NamespacedName{Namespace: deployment.Namespace, Name: deployment.Name}, deployment),
-			OnContinue(DeleteAction(deployment))),
 		HandleResult(
 			GetAction(types.NamespacedName{Namespace: prom.Namespace, Name: prom.Name}, prom),
 			OnContinue(DeleteAction(prom))),
 	)
 }
 
+func (r *MeterBaseReconciler) installPrometheusServingCertsCABundle(
```

Comment:
factory is apart of the MeterBaseReconciler - we can refactor these functions to use that. The original code passed it in functions but if its' in a struct we can clean this function up a bit.

Reply:
updated this func, as well as the others to use r.factory